src/fsharp/TastPickle.fs

```diff
@@ -1305,11 +1304,27 @@ let u_ILInstr st =
 // Pickle/unpickle for F# types and module signatures
 //---------------------------------------------------------------------------
 
-let p_Map pk pv = p_wrap Map.toList (p_list (p_tup2 pk pv))
+let p_Map_core pk pv (xs: Map<_, _>) st =
+    for x in xs do
```

Comment:
`Map.iter` is a better bet I think.

Reply:
(`Map.iter` is I think considerably faster than creating a Map IEnumerator and iterating it)

Reply:
Gotcha, makes sense.